src/buildercore/cfngen.py

```diff
@@ -185,6 +186,10 @@ def build_context_cloudfront(context, parameterize):
             'headers': context['project']['aws']['cloudfront']['headers'],
             'default-ttl': context['project']['aws']['cloudfront']['default-ttl'],
             'errors': errors,
+            'origins': OrderedDict([
+                (o_id, {'hostname': parameterize(o['hostname']), 'pattern': o.get('pattern')})
+                for o_id, o in context['project']['aws']['cloudfront']['origins'].iteritems()
+            ]),
```

Comment:
this is a little gnarly on first blush, but I get it now after a few reads. I (personally) would have done a slightly more verbose version like:

```
def tf(oitem):
    o_id, o = oitem
    return (o_id, {
        'hostname': parameterize(o['hostname']), 
        'pattern': o.get('pattern')
    }
OrderedDict(map(tf, context['project']['aws']['cloudfront']['origins'].items()))
```

but it's fine. pretty sure my inline functions are just as incomprehensible sometimes, too. Stick with `.items()` though. No need for an iterator there. In Python 3 there is no more `iteritems` as `items` becomes lazy.

I thought you might have had a problem assuming the order of the origins are preserved, but I'm that yaml is read in using `buildercore.utils.ordered_load`, so there shouldn't be a problem there either.